### PR TITLE
enable nullable in MSBuildWorkspace

### PR DESCRIPTION
**Pull request by jmarolf**

### CODE REVIEW

File: src/Workspaces/Core/MSBuild/MSBuild/MSBuildProjectLoader.Worker_ResolveReferences.cs [hunk 1]
```diff
@@ -349,8 +346,15 @@ private async Task<bool> VerifyProjectOutputExistsAsync(string projectPath, Reso
 
                 foreach (var projectFileInfo in projectFileInfos)
                 {
+                    if (projectFileInfo.IsEmpty)
+                    {
+                        continue;
+                    }
```
Comment (Youssef1313) [inline src/Workspaces/Core/MSBuild/MSBuild/MSBuildProjectLoader.Worker_ResolveReferences.cs:352]: Does this make any difference?

File: src/Workspaces/Core/MSBuild/MSBuild/MSBuildWorkspace.cs [hunk 2]
```diff
@@ -324,6 +324,11 @@ protected override void ApplyProjectChanges(ProjectChanges projectChanges)
                 if (HasProjectFileChanges(projectChanges))
                 {
                     var projectPath = project.FilePath;
+                    if (projectPath is null)
+                    {
+                        _reporter.Report(new ProjectDiagnostic(WorkspaceDiagnosticKind.Failure, $"Project path for '{project.Name}' was null", projectChanges.ProjectId));
```
Comment (JoeRobich) [inline src/Workspaces/Core/MSBuild/MSBuild/MSBuildWorkspace.cs:329]: Diagnostic strings need to be localized.

File: src/Workspaces/Core/MSBuild/MSBuild/MSBuildWorkspace.cs [hunk 3]
```diff
@@ -519,7 +556,7 @@ protected override void ApplyProjectReferenceAdded(ProjectId projectId, ProjectR
             Debug.Assert(_applyChangesProjectFile != null);
 
             var project = this.CurrentSolution.GetProject(projectReference.ProjectId);
-            if (project != null)
+            if (project is not null && project.FilePath is not null)
```
Comment (JoeRobich) [inline src/Workspaces/Core/MSBuild/MSBuild/MSBuildWorkspace.cs:559]:
```suggestion
            if (project?.FilePath is not null)
```

File: src/Workspaces/Core/MSBuild/MSBuild/ProjectFile/CommandLineArgumentReader.cs [hunk 4]
```diff
@@ -45,7 +43,7 @@ protected void Add(string name, string value)
         {
             ValidateName(name);
 
-            if (string.IsNullOrEmpty(value) || value.Contains(char.IsWhiteSpace))
+            if (RoslynString.IsNullOrEmpty(value) || value.Contains(char.IsWhiteSpace))
```
Comment (sharwell) [inline src/Workspaces/Core/MSBuild/MSBuild/ProjectFile/CommandLineArgumentReader.cs:46]: 📝 `value` allows null

**Review by sharwell — COMMENTED** [review]
Done with 28/42 files so far

File: src/Workspaces/Core/MSBuild/MSBuild/Constants/TargetNames.cs [hunk 5]
```diff
@@ -2,8 +2,6 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#nullable disable
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/MSBuild/MSBuild/Constants/TargetNames.cs:5]: nice

File: src/Workspaces/Core/MSBuild/MSBuild/MSBuildProjectLoader.Worker_ResolveReferences.cs (hunk 1 above)
Comment (jmarolf) [reply src/Workspaces/Core/MSBuild/MSBuild/MSBuildProjectLoader.Worker_ResolveReferences.cs:352]: if the file info is empty then we expect all path information (`OutputFilePath` and `OutputRefFilePath`) to be `null`

File: src/Workspaces/Core/MSBuild/MSBuild/MSBuildProjectLoader.Worker_ResolveReferences.cs [hunk 6]
```diff
@@ -349,6 +346,7 @@ private async Task<bool> VerifyProjectOutputExistsAsync(string projectPath, Reso
 
                 foreach (var projectFileInfo in projectFileInfos)
                 {
+
```
Comment (JoeRobich) [inline src/Workspaces/Core/MSBuild/MSBuild/MSBuildProjectLoader.Worker_ResolveReferences.cs:349]:
```suggestion
```

**Review by JoeRobich — APPROVED** [review]
Looks good to me

**Pull request merged**